src/Hyperion/Extensions/TypeEx.cs

```diff
@@ -192,12 +193,14 @@ private static bool UnsafeInheritanceCheck(Type type)
             return false;
         }
         
-        public static Type LoadTypeByName(string name, bool disallowUnsafeTypes)
+        public static Type LoadTypeByName(string name, bool disallowUnsafeTypes, ITypeFilter typeFilter)
         {
-            if (disallowUnsafeTypes && UnsafeTypesDenySet.Any(name.Contains))
+            if (disallowUnsafeTypes)
             {
-                throw new EvilDeserializationException(
-                    "Unsafe Type Deserialization Detected!", name);
+                if(UnsafeTypesDenySet.Any(name.Contains))
```

Comment:
How do we safely check if a type is not custom type designed to bypass this guard? Also, if `System.Security.Claims.ClaimsIdentity` is contained in `UnsafeTypesDenySet` `name.Contains(Claims.ClaimsIdentity)` will will return true, right?

Reply:
Ok